OpenRA.Mods.Cnc/Traits/ConyardChronoReturn.cs

```diff
@@ -0,0 +1,251 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.GameRules;
+using OpenRA.Mods.Common;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Mods.Common.Traits.Render;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Cnc.Traits
+{
+	[Desc("Implements the special case handling for the Chronoshiftable return on a construction yard.",
+		"Actors that are actively (un)deploying will be returned to the origin as the original actor.",
+		"Otherwise, a vortex animation is played and damage is dealt each tick, ignoring modifiers.")]
+	public class ConyardChronoReturnInfo : ITraitInfo, Requires<HealthInfo>, Requires<WithSpriteBodyInfo>
+	{
+		[Desc("Sequence name with the baked-in vortex animation"), SequenceReference]
+		public readonly string Sequence = "pdox";
+
+		[Desc("Sprite body to play the vortex animation on.")]
+		public readonly string Body = "body";
+
+		[GrantedConditionReference]
+		[Desc("Condition to grant while the vortex animation plays.")]
+		public readonly string Condition = null;
```

Comment:
You are setting defaults for `Sequence` and `Body` that are pretty specific to this use case. Any reason you didn't include values for `Condtion`, `Damage` and `OriginalActor` as default values here?

Reply:
The condition and damage are to aid with self-documentation.